File: python/724.find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left, right = 0, sum(nums[i] for i in range(1, len(nums)))

        if left == right:
            return 0

        for i in range(1, len(nums)):
            left += nums[i - 1]
            right -= nums[i]

            if left == right:
                return i

        return -1
        

File: golang/724.find-pivot-index.go
func pivotIndex(nums []int) int {
    left, right := 0, 0
    for i := 1; i < len(nums); i++ { right += nums[i] }

    if left == right {
        return 0
    }

    for i := 1; i < len(nums); i++ {
        left += nums[i - 1]
        right -= nums[i]
        if left == right {
            return i
        }
    }

    return -1
}